### PR TITLE
SPD changes for Trusty and TLKD

### PR DESCRIPTION
**Pull request by vwadekar**
SPD changes from the downstream repository

### CODE REVIEW

**Comment by davidcunado-arm** [comment]
jenkins: test this please

File: include/lib/aarch64/smcc_helpers.h [hunk 1]
```diff
@@ -56,6 +56,22 @@
...
+#define SMC_RET5(_h, _x0, _x1, _x2, _x3, _x4)	{		\
+	write_ctx_reg(get_gpregs_ctx(_h), CTX_GPREG_X4, (_x4));	\
+	SMC_RET4(_h, (_x0), (_x1), (_x2), (_x3));		\
+}
+#define SMC_RET6(_h, _x0, _x1, _x2, _x3, _x4, _x5)	{	\
+	write_ctx_reg(get_gpregs_ctx(_h), CTX_GPREG_X5, (_x5));	\
+	SMC_RET5(_h, (_x0), (_x1), (_x2), (_x3), (_x4));	\
+}
+#define SMC_RET7(_h, _x0, _x1, _x2, _x3, _x4, _x5, _x6)	{	\
+	write_ctx_reg(get_gpregs_ctx(_h), CTX_GPREG_X6, (_x6));	\
```
Comment (davidcunado-arm) [inline include/lib/aarch64/smcc_helpers.h]: Minor update - could you align the \ for SMC_RET7 and SMC_RET8?

**Comment by antonio-nino-diaz-arm** [comment]
The failure is a false positive in the static checks due to a misbehaving script.

File: services/spd/trusty/trusty.c [hunk 2]
```diff
@@ -414,6 +414,14 @@ static int32_t trusty_setup(void)
...
 					    DAIF_ABT_BIT);
 
+	/*
+	 * arg0 = TZDRAM aperture available for BL32
+	 * arg1 = BL32 boot params
+	 * arg2 = BL32 boot params length
+	 */
+	ep_info->args.arg1 = ep_info->args.arg2;
+	ep_info->args.arg2 = 4096*2;
+
```
Comment (davidcunado-arm) [inline services/spd/trusty/trusty.c]: This change doesn't seem to match the commit message.
Also, would it be better to have a constant for the params length?

File: include/lib/aarch64/smcc_helpers.h (hunk 1 above)
Comment (vwadekar) [inline include/lib/aarch64/smcc_helpers.h]: Yes. Will update the PR

Comment (davidcunado-arm) [inline include/lib/aarch64/smcc_helpers.h]: Minor update - Could align the \ on line 68 with the one on line 67?

File: services/spd/tlkd/tlkd_main.c [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2015, ARM Limited and Contributors. All rights reserved.
+ * Copyright (c) 2015-2016, ARM Limited and Contributors. All rights reserved.
```
Comment (davidcunado-arm) [inline services/spd/tlkd/tlkd_main.c]: Update copyright date to 2017.

**Comment by vwadekar** [comment]
@davidcunado-arm 

The alignment shows up properly in my editor. What is the exact issue?

File: services/spd/tlkd/tlkd_main.c (hunk 3 above)
Comment (vwadekar) [inline services/spd/tlkd/tlkd_main.c]: This patch was created in 2016. I am just pushing it upstream. Any specific reason to update it to 2017?

Comment (davidcunado-arm) [inline services/spd/tlkd/tlkd_main.c]: In that case, no I don't think so - I was assuming that this was a new patch as one of the other files had a 2017 date.

**Comment by davidcunado-arm** [comment]
@vwadekar 
Ok, just looked at the change in my text editor - actually looks ok compared to what the github diff tool is showing, but with exception of line 73. 
Ideally, the \ on lines 71 and 72 would aligned with that of line 73.

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Pull request merged**